test/functional/provider/perl_spec.lua

```diff
@@ -0,0 +1,80 @@
+local helpers = require('test.functional.helpers')(after_each)
+local eq, clear = helpers.eq, helpers.clear
+local missing_provider = helpers.missing_provider
+local command = helpers.command
+local write_file = helpers.write_file
+local eval = helpers.eval
+local retry = helpers.retry
+
+do
+  clear()
+  local reason = missing_provider('perl')
+  if reason then
+    pending(string.format("Missing perl host, or perl version is too old (%s)", reason), function() end)
+    return
+  end
+end
+
+before_each(function()
+  clear()
+end)
+
+describe('perl host', function()
+  if helpers.pending_win32(pending) then return end
```

Comment:
https://github.com/neovim/neovim/pull/11595#discussion_r361154077
stdin polling shouldn't be a blocker for these tests on Windows since the host is not running but these tests always fail. My guess is the way I used `socket_session` to create a new session.